fork-common/src/test/java/com/shazam/fork/model/TestCaseEventBuilderTest.java

```diff
@@ -1,57 +1,158 @@
+                .withTestClass("testClass")
+                .withTestMethod("testMethod")
                 .withIsIgnored(true)
-                .withTestClass("TestClass")
+                .withProperties(properties())
+                .build();
+
+        TestCaseEvent testCaseEventB = new TestCaseEvent.Builder()
+                .withTestClass("testClass")
                 .withTestMethod("testMethod")
-                .withProperties(properties)
+                .withIsIgnored(false)
+                .withProperties(otherProperties())
                 .build();
+
+        assertEquals(testCaseEventA, testCaseEventB);
+    }
+
+    @Test
+    public void testCaseEventsAreEqualWhenTestClassAndTestMetodAndIgnoreAndPropertiesAreTheSame() {
+        TestCaseEvent testCaseEventA = new TestCaseEvent.Builder()
+                .withTestClass("testClass")
+                .withTestMethod("testMethod")
+                .withIsIgnored(true)
+                .withProperties(properties())
+                .build();
+
         TestCaseEvent testCaseEventB = new TestCaseEvent.Builder()
+                .withTestClass("testClass")
+                .withTestMethod("testMethod")
                 .withIsIgnored(true)
-                .withTestClass("TestClass")
+                .withProperties(properties())
+                .build();
+
+        assertEquals(testCaseEventA, testCaseEventB);
+    }
+
+    @Test
+    public void testCaseEventsAreNotEqualWhenTestClassIsNotTheSame() {
+        TestCaseEvent testCaseEventA = new TestCaseEvent.Builder()
+                .withTestClass("testClass")
                 .withTestMethod("testMethod")
-                .withProperties(properties)
                 .build();
 
-        assertThat(testCaseEventA, equalTo(testCaseEventB));
+        TestCaseEvent testCaseEventB = new TestCaseEvent.Builder()
+                .withTestClass("anothertestClass")
+                .withTestMethod("TestMethod")
+                .build();
+
+        assertNotEquals(testCaseEventA, testCaseEventB);
     }
 
     @Test
-    public void twoObjectsWithDifferentParametersShouldNotBeEqual() {
-        HashMap<String, String> propertiesA = new HashMap<>();
-        propertiesA.put("key", "value");
+    public void testCaseEventsAreNotEqualWhenTestMethodIsNotTheSame() {
+        TestCaseEvent testCaseEventA = new TestCaseEvent.Builder()
+                .withTestClass("testClass")
+                .withTestMethod("testMethod")
+                .build();
 
-        HashMap<String, String> propertiesB = new HashMap<>();
-        propertiesB.put("key2", "value2");
+        TestCaseEvent testCaseEventB = new TestCaseEvent.Builder()
+                .withTestClass("testClass")
+                .withTestMethod("anotherTestMethod")
+                .build();
+
+        assertNotEquals(testCaseEventA, testCaseEventB);
+    }
 
+    @Test
+    public void testCaseEventsHaveSameHasheCodeWhenTestClassAndTestMetodAreTheSame() {
```

Comment:
Typo in the name 
testCaseEventsHaveSameHasheCodeWhenTestClassAndTestMetodAreTheSame
->
testCaseEventsHaveSameHashCodeWhenTestClassAndTestMethodAreTheSame